haystack/preview/components/joiners/join.py

```diff
@@ -0,0 +1,37 @@
+from typing import Type
+from haystack.preview import component, default_from_dict, default_to_dict
+
+
+@component
+class Join:
+    """
+    Simple component that joins together a group of inputs of the same type. Works with every type that supports
+    the + operator for joining, like lists, strings, etc.
+    """
+
+    def __init__(self, inputs_count: int, inputs_type: Type):
+        """
+        :param inputs_count: the number of inputs to expect.
+        :param inputs_type: the type of the inputs. Every type that supports the + operator works.
+        """
+        self.inputs_count = inputs_count
```

Comment:
Why not set it to 2 by default?

Reply:
Why having `inputs_count` at all? The degenerate case would be having only one input, that would make the component a no-op

Reply:
Sorry @masci I don't get your point. I can set it to be minimum two, but in principle being able to set how many inputs to expect seems way more useful than fixing it to two. There are many cases where you want to aggregate several values (just look at (this pipeline)[https://github.com/deepset-ai/haystack/issues/5614])

Reply:
> Why having inputs_count at all?
I'm not saying to hardcode to two, I'm talking about removing it, why do we need an upper limit?

Reply:
This is not an upper limit: you need to specify in advance how many inputs you expect. I mean, I could set it to a huge number and make them all optional, but it will make quite some noise in the error messages: if you fail to connect it, the error message will list all the possible input connections and the error becomes seriously unreadable :sweat_smile:  I'd rather let the user specify how many they want and stick to that. It makes debugging easier.

If this is a big deal let's discuss it offline.

Reply:
Yeah, it would be great to remove `input_counts` parameter if possible.